cmd/gapir/cc/main.cpp

```diff
@@ -395,8 +536,134 @@ void android_process(struct android_app* app, int32_t cmd) {
 
 }  // namespace
 
+// Extract command line arguments from the extra of Android intent:
+//   adb shell am start -n <...> -e gapir-launch-args "'list of arguments to be
+//   extracted'"
+// Note the quoting: from host terminal adb command, we need to double-escape
+// the extra args string, as it is first quoted by host terminal emulator
+// (e.g. bash), then it must be quoted for the on-device shell.
+void GetArgvFromIntents(struct android_app* state, int* argc, char*** argv) {
```

Comment:
I won't hold up the CR for this, but I have a preference of making this slightly more C++-ey.
```
struct Args {
   int argc;
   char** argv;
   - Other stuff here?
   ~Args() { 
      - free argv here
   }
}
```

You can return by std::move(), so that we don't have to heap-alloc the struct itself.

Reply:
This way no-one has to forget to "Free"

Reply:
Isn't this dangerous in its own right; that anyone returning Args who forgets to use std::move will return a pointer into freed memory?

Reply:
OK, changed this code so it uses a std::vector<std::string> for data storage and string::c_str() / std::vector<unsigned  char*> to create a presentation of that data which looks like old fashioned c strings.

I think this is the most "c++ey" way of solving the problem.

God knows why anyone's still using c strings in c++.

Reply:
As for "anyone who forgets to use std::move" we can disable copy construction/copy assignment. Then they will just get a compile-error in this case.

Reply:
What's your opinion of the std::string proposal I uploaded in the last revision?

Reply:
That being said how you restructured it is completely fine with me.

We typically use c strings in 2 cases.
1) When we don't really have a choice. (Interfacing with a C api. eg. `int main(....))`
2) When we want to avoid copies. std::string can be a pain for a number of reasons, and often we want to avoid copies. C++14 doesn't yet have string_view or equivalent, and we haven't pulled an implementation in yet.

But for most other cases raw cstrings should generally be avoided.

Reply:
Yep the last version works for me. Thanks!